### DISCUSSION TITLE
Installation on MAC M2

### DISCUSSION THREAD

**Discussion by ozarchie** [gusmanb/logicanalyzer#75] [Q&A] [answered]
Do you have a simple step by step for installation on the Apple ARM based computers?
Is there sample data available to practice with while waiting for hardware?

**Comment by gusmanb** [comment]
Hi.

[Here](https://github.com/gusmanb/logicanalyzer/wiki/04---LogicAnalyzer-Software#notes-for-linux-and-macos-users) you can find instructions on how to make it run on MacOS for ARM devices.

Regarding sample data, here are two example captures, one from an I2C device and other from a SPI one: [sample_captures.zip](https://github.com/gusmanb/logicanalyzer/files/12406576/sample_captures.zip)

Cheers.

**Comment by ozarchie** [comment, marked as answer]
G'day,
Thanks for your quick response and for a great project.
I am trying V5.

My process for running on MACBook M2 silicon was:

1. Unzip the release
2. sudo xattr -r -d com.apple.quarantine osx-arm64
3. cd osx-arm64
4. chmod +x LogicAnalyzer
5. ./LogicAnalyzer

But the outcome was: 
zsh: killed     ./LogicAnalyzer
So the OS killed the LogicAnalyzer, as reported by zsh, but with no details.

I installed an earlier version but - same result.

I found a solution using: (https://github.com/dotnet/runtime/issues/79267)

1. Unzip the release
2. sudo xattr -r -d com.apple.quarantine osx-arm64
3. cd osx-arm64
4. chmod +x LogicAnalyzer
5. codesign --force --deep -s - ./LogicAnalyzer
6. ./LogicAnalyzer

I was able to run and load the samples that you sent.
Thankyou for your help. I will give more feedback when the pico and pico-w boards arrive.
Cheers
Archie

**Comment by ozarchie** [comment]
Hi,
I have installed VS2022 and have now been able to compile using VSCODE.
All went well and the application runs on the M2 silicon without a problem.
Cheers
Archie
